geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/filters/DtgAgeOffFilter.scala

```diff
@@ -100,10 +104,52 @@ object DtgAgeOffFilter {
         i
     }
 
+    // Handle changing dtgIndex for join indexes
+    val updatedDtgIndex = {
+      if (index.name.contains("join")) {
+        val joinSft = buildIndexSft(sft)
+        joinSft.indexOf(sft.getDescriptor(dtgIndex).getLocalName)
+      } else {
+        dtgIndex
+      }
+    }
+
     AgeOffFilter.configure(sft, expiry) ++ Map (
       Configuration.SftOpt   -> SimpleFeatureTypes.encodeType(sft),
       Configuration.IndexOpt -> index.identifier,
-      Configuration.DtgOpt   -> dtgIndex.toString
+      Configuration.DtgOpt   -> updatedDtgIndex.toString
     )
   }
+
+  // From IndexValueEncoder
```

Comment:
@elahrvivaz since these are copied from IndexValueEncoder, is there a good place to relocate them? 

Also, since they are private in the other location, moving them to the geomesa-index-api module should be (api-)safe, right?